File: KIWI/Service/ConversionDAO.vb
Imports IDAUtil
Imports [lib]

Public Class ConversionDAO

    Private ReadOnly dbServer As IDBServerConnector

    Public Sub New(dbServer As IDBServerConnector)
        Me.dbServer = dbServer
    End Sub

    Public Function getQuantityConversionMaterialList() As List(Of ConversionMaterial)
        dbServer.createConnectionToServerIniCF()
        Dim conversionMaterialList As List(Of ConversionMaterial)
        conversionMaterialList = dbServer.getObjectListByQuery(Of ConversionMaterial)("SELECT * FROM QuantityConversionMaterial")
        dbServer.closeConnection()
        Return conversionMaterialList
    End Function

    Public Function getQuantityConversionShipToList() As List(Of ConversionShiptTo)
        dbServer.createConnectionToServerIniCF()
        Dim conversionShipToList As List(Of ConversionShiptTo)
        conversionShipToList = dbServer.getObjectListByQuery(Of ConversionShiptTo)("SELECT * FROM QuantityConversionShipTo")
        dbServer.closeConnection()
        Return conversionShipToList
    End Function

    Public Function getQuantityConversionLogList(fromDocument As Long, toDocument As Long) As List(Of ConversionLog)
        dbServer.createConnectionToServerIniCF()
        Dim conversionLogList As List(Of ConversionLog)
        conversionLogList = dbServer.getObjectListByQuery(Of ConversionLog)($"SELECT * FROM QuantityConversionLog WHERE orderNumber >= {fromDocument} AND orderNumber <= {toDocument}")
        dbServer.closeConnection()
        Return conversionLogList
    End Function
End Class

File: KIWI/Service/ConversionDAO.cs
using lib;
using System.Collections.Generic;

namespace KIWI {
    public class ConversionDAO {
        private readonly IDBServerConnector dbServer;

        public ConversionDAO(IDBServerConnector dbServer) {
            this.dbServer = dbServer;
        }

        public List<ConversionMaterial> getQuantityConversionMaterialList() {
            dbServer.createConnectionToServerCF();
            List<ConversionMaterial> conversionMaterialList;
            conversionMaterialList = dbServer.getObjectListByQuery<ConversionMaterial>("SELECT * FROM QuantityConversionMaterial");
            dbServer.closeConnection();
            return conversionMaterialList;
        }

        public List<ConversionShiptTo> getQuantityConversionShipToList() {
            dbServer.createConnectionToServerCF();
            List<ConversionShiptTo> conversionShipToList;
            conversionShipToList = dbServer.getObjectListByQuery<ConversionShiptTo>("SELECT * FROM QuantityConversionShipTo");
            dbServer.closeConnection();
            return conversionShipToList;
        }

        public List<ConversionLog> getQuantityConversionLogList(long fromDocument, long toDocument) {
            dbServer.createConnectionToServerCF();
            List<ConversionLog> conversionLogList;
            conversionLogList = dbServer.getObjectListByQuery<ConversionLog>($"SELECT * FROM QuantityConversionLog WHERE orderNumber >= {fromDocument} AND orderNumber <= {toDocument}");
            dbServer.closeConnection();
            return conversionLogList;
        }
    }
}